account_invoice_inter_company/models/account_invoice.py

```diff
@@ -0,0 +1,276 @@
+            force_number = inter_invoice.move_name
+            inter_invoice.move_name = False
+            inter_invoice.unlink()
+        # create invoice
+        dest_invoice_data = self._prepare_invoice_data(dest_company)
+        if force_number:
+            dest_invoice_data['move_name'] = force_number
+        dest_invoice = self.create(dest_invoice_data)
+        # create invoice lines
+        for src_line in self.invoice_line_ids:
+            if not src_line.product_id:
+                raise UserError(_(
+                    "The invoice line '%s' doesn't have a product. "
+                    "All invoice lines should have a product for "
+                    "inter-company invoices.") % src_line.name)
+            dest_inv_line_data = src_line._prepare_invoice_line_data(
+                dest_invoice, dest_company)
+            self.env['account.invoice.line'].create(dest_inv_line_data)
+        # add tax_line_ids in created invoice
+        dest_invoice_line_ids = dest_invoice.invoice_line_ids
+        if (any(
+            line.invoice_line_tax_ids for line in dest_invoice_line_ids) and
+                not dest_invoice.tax_line_ids):
+            dest_invoice.compute_taxes()
+        # Validation of account invoice
+        precision = self.env['decimal.precision'].precision_get('Account')
+        if (dest_company.invoice_auto_validation and
+                not float_compare(self.amount_total,
+                                  dest_invoice.amount_total,
+                                  precision_digits=precision)):
+            dest_invoice.action_invoice_open()
+        else:
+            # Add warning in chatter if the total amounts are different
+            if float_compare(self.amount_total, dest_invoice.amount_total,
+                             precision_digits=precision):
+                body = (_(
+                    "WARNING!!!!! Failure in the inter-company invoice "
+                    "creation process: the total amount of this invoice "
+                    "is %s but the total amount of the invoice %s "
+                    "in the company %s is %s")
+                    % (dest_invoice.amount_total, self.number,
+                       self.company_id.name, self.amount_total))
+                dest_invoice.message_post(body=body)
+        return {'dest_invoice': dest_invoice}
+
+    @api.multi
+    def _get_destination_invoice_type(self):
+        self.ensure_one()
+        MAP_INVOICE_TYPE = {
+            'out_invoice': 'in_invoice',
+            'in_invoice': 'out_invoice',
+            'out_refund': 'in_refund',
+            'in_refund': 'out_refund',
+        }
+        return MAP_INVOICE_TYPE.get(self.type)
+
+    @api.multi
+    def _get_destination_journal_type(self):
+        self.ensure_one()
+        MAP_JOURNAL_TYPE = {
+            'out_invoice': 'purchase',
+            'in_invoice': 'sale',
+            'out_refund': 'purchase_refund',
```

Comment:
In 'out_refund' should be 'purchase' and in 'in_refund' should be 'sale', as since v9 this has changed in odoo. Check this link the journal type in odoo v11: https://github.com/odoo/odoo/blob/11.0/addons/account/models/account.py#L320

@jbeficent

Reply:
Thank you @rgarnau ! This issue must be fixed for the module in V10 as well!

Reply:
and v9!

Reply:
I didn' find this module for V9, but if you know where to find it just send me the link and I will create a PR for V9 as well ;)

Reply:
it's true, it doesn't exist for v9